backend/satellite_tools/reposync.py

```diff
@@ -1153,7 +1153,14 @@ def import_packages(self, plug, source_id, url, is_non_local_repo):
                         os.remove(stage_path)
                     if os.path.exists(os.path.dirname(stage_path)):
                         # remove the checksum directory
-                        os.rmdir(os.path.dirname(stage_path))
+                        try:
+                            os.rmdir(os.path.dirname(stage_path))
+                        except OSError:
+                            exc = sys.exc_info()[1]
+                            if exc.errno == errno.ENOTEMPTY:
+                                pass
```

Comment:
This should go to the log error instead.

Reply:
No, why?

It is an expected condition and there is nothing to do in that case. We just skip the directory, will be cleaned up later.

Reply:
Ah right. Then let's say it so:

```python
try:
    ...
except OSError as exc:
    if exc.errno != errno.ENOEMPTY:
        raise
```

Reply:
Ah right. Then let's say it so:

```python
try:
    ...
except OSError as exc:
    if exc.errno != errno.ENOEMPTY:
        raise
```

Reply:
I prefer the slightly pedantic, but more explicit version. @mcalmer, can you break the tie?

Reply:
What about a mix of both:

```python
try:
    ...
except OSError as exc:
    if exc.errno == errno.ENOTEMPTY:
        pass
    else:
        raise exc
```

It is explicit and do not use this dirty `sys.exc_info()[1]` which nobody understand :-)

Reply:
thanks

Reply:
No, this pass is redundant and absolutely unnecessary here.